cli/types/cypress.d.ts

```diff
@@ -3066,7 +3066,12 @@ declare namespace Cypress {
    * Takes ComponentDevServerOpts to track the signature of the devServerConfig for the provided `devServer`,
    * so we have proper completion for `devServerConfig`
    */
-  type ConfigOptions<ComponentDevServerOpts = any> = Partial<UserConfigOptions<ComponentDevServerOpts>>
+  type ConfigOptions<ComponentDevServerOpts = any> = Partial<UserConfigOptions<ComponentDevServerOpts>> & {
+    /**
+     * Hosts mappings to IP addresses.
+     */
+     hosts?: null | {[key: string]: string}
+  }
```

Comment:
This should be added here because we can add `hosts` to `definedConfig` like `driver/cypress.config.ts`.

Reply:
If we have `hosts?`, do we also need `null | {...}`. Feels like we should have undefined or a value, rather than undefined, null or a value; Is there a difference between null and undefined here? (and if there is, why would we do that >.> )

Reply:
I copied the type from below:

https://github.com/cypress-io/cypress/blob/3d98f9813619af5485ba6aef03ec596deaacfa6f/cli/types/cypress.d.ts#L2950-L2953

As it is inside `RuntimeConfigOptions`, I think they're the same type.